paderbox/array/interval/core.py

```diff
@@ -466,33 +474,66 @@ def __setitem__(self, item, value):
         >>> ai
         ArrayInterval("0:10, 40:50", shape=(50,))
 
+        >>> ai = zeros(50)
+        >>> ai2 = zeros(10)
+        >>> ai2[5:10] = 1
+        >>> ai[10:20] = ai2
+        >>> ai
+        ArrayInterval("15:20", shape=(50,))
+        >>> ai2 = zeros(20)
+        >>> ai2[0:10] = 1
+        >>> ai2[15:20] = 1
+        >>> ai[10:30] = ai2
+        >>> ai
+        ArrayInterval("10:20, 25:30", shape=(50,))
+
+        >>> ai[40:60] = ai2
+        Traceback (most recent call last):
+            ...
+        IndexError: Index 60 out of bounds for ArrayInterval with size 50
         """
+        if not isinstance(item, slice):
+            raise NotImplementedError(
+                f'{self.__class__.__name__}.__setitem__ only supports slices '
+                f'for indexing, not {item}'
+            )
 
         start, stop = cy_parse_item(item, self.shape)
 
         if np.isscalar(value):
+            if value not in (0, 1):
+                # Numpy interprets values as boolean even if they are larger
+                # than 1, why doesn't the ArrayInterval do the same?
+                raise ValueError(value)
+            value = bool(value)
             if self.inverse_mode:
-                if value == 0:
-                    self.intervals = self.intervals + ((start, stop),)
-                elif value == 1:
-                    self.intervals = cy_non_intersection((start, stop), self.intervals)
-                else:
-                    raise ValueError(value)
+                value = not value
+            if value:
+                self.intervals = self.intervals + ((start, stop),)
             else:
-                if value == 1:
-                    self.intervals = self.intervals + ((start, stop),)
-                elif value == 0:
-                    self.intervals = cy_non_intersection((start, stop), self.intervals)
-                else:
-                    raise ValueError(value)
-        elif isinstance(value, (tuple, list, np.ndarray)):
-            assert len(value) == stop - start, (start, stop, stop - start, len(value), value)
-            ai = ArrayInterval(value, inverse_mode=self.inverse_mode)
-            intervals = self.intervals
-            intervals = cy_non_intersection((start, stop), intervals)
-            self.intervals = intervals + tuple([(s + start, e + start) for s, e in ai.intervals])
+                self.intervals = cy_non_intersection((start, stop), self.intervals)
+        elif isinstance(value, (tuple, list, np.ndarray, ArrayInterval)):
+            if not isinstance(value, ArrayInterval):
+                value = ArrayInterval(value, inverse_mode=self.inverse_mode)
```

Comment:
Can you add a comment, why `inverse_mode=self.inverse_mode`?

Reply:
Done